Libraries/Connect/Interfaces/MethodDescriptor.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Contains metadata on a specifific RPC method.
```

Comment:
I'd lean away from calling this "MethodDescriptor" - it sounds like we're building support for descriptor-based dynamic protobuf messages, and this is a much more lightweight type by design. `connect-go` calls this a [Spec](https://pkg.go.dev/github.com/bufbuild/connect-go#Spec), short for specification - would that work here? It also seems useful to me to include  the RPC type, since it's useful in logging/tracing and deciding on a reasonable timeout.

Final nit: spelling of "specific"

Reply:
Fair enough, I'm ok with renaming to `Spec` since that will nicely match connect-go. The original naming was based on what [`grpc-swift` does](https://github.com/grpc/grpc-swift/blob/bcca31f3cfe1a20ca598e14b4e3663206a449813/Sources/Examples/Echo/Model/echo.grpc.swift#L399-L423). I can also add the RPC type in the same way connect-go does.